### PR TITLE
User story 14

### LINKED ISSUE #14: User Story 20, User can delete an author
```
As a Visitor,
When I visit an author's show page,
I see a link on the page to delete the author.
This link should return me to the book index page where I
no longer see this author listed.
If this author was the only author for any book, that book is also deleted.
If this author co-authored a book with someone else, that book should also be deleted, but not the other author.

(you may need to delete other content before you can delete an author or book)
```

DO THESE STORIES FIRST:
- User Story 14, Author Show Page

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [x] edge case ("sad path") tests written
- [x] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #9

child of #33

depends on #14

### LINKED ISSUE #28: User Story 14, Author Show Page
```
As a Visitor,
When I visit an author's show page
I see all books by that author
Each book should show:
- the book title
- the number of pages in the book
- the year the book was published
- a small image of the book cover

Each book should show a list of any other authors
(exclude this show page's author from that list)
```

DO THESE STORIES FIRST:
- User Story 12, Add a New Book to the System

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [x] edge case ("sad path") tests written
- [x] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #15

child of #33

depends on #17

### PR DESCRIPTION
**Pull request by chakeresa**
This PR/branch...
 - Resolves #28  (User Story 14 - author show page)
 - Auto-adds GET routes for authors#show & authors#index
 - Adds the author show page and its contents
 - Adds Book#coauthors method
 - Tests everything added

### CODE REVIEW

File: app/controllers/authors_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class AuthorsController < ApplicationController
+  def show
+    if Author.pluck(:id).include?(params[:id].to_i)
+      @author = Author.find(params[:id])
+    else
+      redirect_to authors_path
+    end
+  end
+
+  def index
```
Comment (DavisC0801) [inline app/controllers/authors_controller.rb:10]: Author index is not needed as per spec.

File: app/controllers/authors_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class AuthorsController < ApplicationController
+  def show
+    if Author.pluck(:id).include?(params[:id].to_i)
+      @author = Author.find(params[:id])
+    else
+      redirect_to authors_path
```
Comment (DavisC0801) [inline app/controllers/authors_controller.rb]: Change redirect to book index page

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   resources :books, only: [:index, :show]
+
+  resources :authors, only: [:index, :show]
```
Comment (DavisC0801) [inline config/routes.rb]:
```suggestion
  resources :authors, only: :show
```

Comment (DavisC0801) [reply config/routes.rb]: Remove additional route as only show page is required.

File: app/views/authors/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  <% end %>
+  <% @author.books.each do |book|%>
+    <div id="book-<%= book.id %>">
+      <img src="<%= book.thumbnail %>" alt="Cover of <%= book.title %>">
+      <h3><a href="<%= book_path(book) %>"><%= book.title %></a></h3>
+      <% if book.author_count > 1 %>
+        <p>
+          Co-authored by
+          <% book.coauthors(@author).each_with_index do |coauthor, i| %>
+            <a href="<%= author_path(coauthor) %>"><%= coauthor.name %></a><%= ", " if i < book.author_count - 2 %>
```
Comment (DavisC0801) [inline app/views/authors/show.html.erb:15]: Possible refactor to controller to pass an argument to the view instead of each with index. Unsure of MVC conventions here.

**Review by DavisC0801 — COMMENTED** [review]
This is really, really solid. Thank you!

File: app/controllers/authors_controller.rb (hunk 1 above)
Comment (chakeresa) [reply app/controllers/authors_controller.rb:10]: Deleted - thanks!
